### PR TITLE
Upgrade magnum to stein

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

**Comment by JohnGarbutt** [comment]
I think we should try that in rocky pre-prod first.

File: etc/kayobe/kolla/globals.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ keystone_tag: 7.0.1.10
 kibana_tag: 7.0.1.10
 kolla_toolbox_tag: 7.0.1.10
 logstash_tag: 7.0.1.10
-magnum_tag: 7.0.1.10
+magnum_tag: stein
```
Comment (JohnGarbutt) [inline etc/kayobe/kolla/globals.yml:78]: are we sure RDO have released a new enough version yet?

Comment (JohnGarbutt) [reply etc/kayobe/kolla/globals.yml:78]: ah, looks like 8.1.0:
http://mirror.centos.org/centos/7/cloud/x86_64/openstack-stein/openstack-magnum-api-8.1.0-1.el7.noarch.rpm

**Review by jovial — COMMENTED** [review]
Won't you also need to set `magnum_api_image` and `magnum_conductor_image` if you plan to use the upstream images. Otherwise, it will try and pull it from the local docker registry.

**Comment by brtkwr** [comment]
> Won't you also need to set `magnum_api_image` and `magnum_conductor_image` if you plan to use the upstream images. Otherwise, it will try and pull it from the local docker registry.

In which file would you set this?

**Comment by jovial** [comment]
> > Won't you also need to set `magnum_api_image` and `magnum_conductor_image` if you plan to use the upstream images. Otherwise, it will try and pull it from the local docker registry.
> 
> In which file would you set this?

etc/kayobe/kolla/globals.yml

I think you could use something like:

`magnum_conductor_image: !unsafe "{{ docker_namespace }}/{{ kolla_base_distro }}-{{ magnum_install_type }}-magnum-conductor"`

or maybe `{% raw %}` in place of `!unsafe`.

**Comment by brtkwr** [comment]
@jovial does that^ work? without defining `docker_namespace`, `kolla_base_distro` or `magnum_install_type` anywhere else in the config?

**Comment by brtkwr** [comment]
Accompanying change on preprod here: https://github.com/RSE-Cambridge/cumulus-kayobe-config/pull/89

File: etc/kayobe/kolla/globals.yml [hunk 2]
```diff
@@ -102,3 +102,7 @@ enable_host_ntp: true
 
 # Default size causes ES to fall over very easily
 es_heap_size: "24g"
+
+# Use upstream images for these
+magnum_api_image: !unsafe "{{ docker_namespace }}/{{ kolla_base_distro }}-{{ magnum_install_type }}-magnum-conductor"
```
Comment (jovial) [inline etc/kayobe/kolla/globals.yml]:
```suggestion
magnum_api_image: !unsafe "{{ docker_namespace }}/{{ kolla_base_distro }}-{{ magnum_install_type }}-magnum-api"
```

**Comment by jovial** [comment]
> @jovial does that^ work? without defining `docker_namespace`, `kolla_base_distro` or `magnum_install_type` anywhere else in the config?

I think this will work because this file first gets templated: https://github.com/openstack/kayobe/blob/aea7117764ee3ced0ccb7477c5c962569fa1c926/ansible/kolla-ansible.yml#L236, before being spat out here: https://github.com/openstack/kayobe/blob/aea7117764ee3ced0ccb7477c5c962569fa1c926/ansible/roles/kolla-ansible/templates/globals.yml.j2

This is then consumed by kolla-ansible which does its own templating. The unsafe is to prevent kayobe from templating that variable.

**Review by paulbrowne — APPROVED** [review]

**Comment by paulbrowne** [comment]
Tested in pre-prod, so we can merge this in prod and upgrade.

**Pull request merged**